pkg/process/checks/process.go

```diff
@@ -271,8 +269,13 @@ func (p *ProcessCheck) run(groupID int32, collectRealTime bool) (RunResult, erro
 		p.lastPIDs = append(p.lastPIDs, pid)
 	}
 
-	if sysProbeUtil := p.getRemoteSysProbeUtil(); sysProbeUtil != nil {
-		mergeProcWithSysprobeStats(p.lastPIDs, procs, sysProbeUtil)
+	if p.sysprobeClient != nil && p.sysProbeConfig.ProcessModuleEnabled {
```

Comment:
Do we need to check `p.sysProbeConfig.ProcessModuleEnabled` again? It seems we only create the client `syscfg.NetworkTracerModuleEnabled || syscfg.ProcessModuleEnabled` on line 147

Reply:
This code should only run if the process module is running in system-probe.